src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -2346,14 +2357,35 @@ public boolean hasSymbolText() {
         if (valueTid == null || IonType.SYMBOL != valueTid.type) {
             return false;
         }
-        return valueTid.isInlineable;
+        return valueTid.isInlineable || valueTid == IonTypeID.SYSTEM_SYMBOL_VALUE;
     }
 
     @Override
     public String getSymbolText() {
         if (isEvaluatingEExpression) {
             return macroEvaluatorIonReader.symbolValue().assumeText();
         }
+        if (valueMarker.typeId == IonTypeID.SYSTEM_SYMBOL_VALUE) {
+            long id;
+            if (valueMarker.startIndex == -1) {
+                id = valueMarker.endIndex;
+            } else {
+                id = readFixedUInt_1_1(valueMarker.startIndex, valueMarker.endIndex);
+
+                // FIXME: This is a hack that works as long as our system symbol table doesn't grow to
```

Comment:
I'm fine tracking this in an issue. This should work for the foreseeable future.

Reply:
I will create an issue with relevant perma-links to the code once this PR is merged.